spg/Animal/lib/Animal.pm

```diff
@@ -36,6 +36,59 @@ if you don't export anything, such as for a purely object-oriented module.
 
 =head1 SUBROUTINES/METHODS
 
+=head2 new
+
+=cut
+
+sub new {
+  my ( $class, $name, $colour ) = @_;
+
+  $colour = default_colour() unless defined $colour;
+
+  return bless { name => $name, colour => $colour }, $class;
+
+}
+
+=head2 set_name
+
+=cut
+
+sub set_name {
+  $_[0]->{name} = $_[1];
+  return $_[0];
+}

```

Comment:
I wouldn't usually recommend doing this, but should this defend against attempts at setting the name on the class rather than on an instance? The exercise definition alludes to this with:

```doc
What should you do if you’re asked to set the name or color of a generic animal?
```